### ISSUE TITLE
Downloading large zip makes zip corrupt

### ISSUE THREAD

**Issue by madsi1m** [filesender/filesender#138]
Hi All,

Found a new (or yet to be discovered old) bug with the downloads page.

To reproduce:
1. Upload several files with a total size of around ~20gb. (uploads fine)
2. on the download page, tick at least 5gb worth and download as a zip file (downloads fine)
3. try unzipping the file. (header of a file in the zip will be corrupt)

Note that downloading the files one by one without the zip seems to be fine.

**Comment by monkeyiq** [comment]
I am trying with 20 files which are 1gb each copied from /dev/zero. I can get 5gb (5 files) fine. The md5sum is the same as they are all zero anyway. Maybe I need to use real data, maybe larger files. I'll keep trying.

**Comment by monkeyiq** [comment]
Excellent, so it happens for files of zero bytes too it seems.
```
$ unzip  transfer.zip 
Archive:  transfer.zip
 extracting: gb8                     
 extracting: gb7                     
 extracting: gb6                     
 extracting: gb5                     
 extracting: gb4                     
file #6:  bad zipfile offset (local header sig):  947913089
file #7:  bad zipfile offset (local header sig):  1996489166
file #8:  bad zipfile offset (local header sig):  3045065243
```

TOC seems somewhat ok, extract isn't though...

```
$ unzip -l transfer.zip 
Archive:  transfer.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
1048576000  10-31-2012 15:09   gb8
1048576000  10-31-2012 15:09   gb7
1048576000  10-31-2012 15:09   gb6
1048576000  10-31-2012 15:09   gb5
1048576000  10-31-2012 15:09   gb4
1048576000  10-31-2012 15:09   gb3
1048576000  10-31-2012 15:09   gb2
1048576000  10-31-2012 15:09   gb1
---------                     -------
8388608000                     8 files
```

**Comment by monkeyiq** [comment]
This appears to be a bug in the zip64 handling code. It will expose for files at offsets over 2^32 bytes in, so you want 5 or 6 one gig files in the zip to trigger it. 

I have tried looking at various extension blocks where the 8 byte offset lives, the best I've made so far is a zip that can expand ok but the tools complain about the format of the headers still.

**Comment by madsi1m** [comment]
Could we try another zip lib? https://github.com/maennchen/ZipStream-PHP might work

**Comment by monkeyiq** [comment]
It is probably a good idea to migrate to a library. It will be easier to maintain and we can always switch libraries later if we want.

Unfortunately the above one is php 7.0+, currently the CI is running in php 5.3 so we would need to have some workaround to also not do zip stuff until the CI env is updated.

**Comment by monkeyiq** [comment]
the larger concern is making sure the zip library can stream from data offered by code. So we can read and stream out the chunked or otherwise handled storage_types. If there is no such support then we would have to read the data into a secure tmpfile and add/stream that to the client for each file.

**Comment by monkeyiq** [comment]
zipstream-php has addFileFromStream() but then we need a stream wrapper for filesender's storage types.

**Comment by monkeyiq** [comment]
@madsi1m I've just merged a fix for this. Hopefully working zips for all!

**Comment by monkeyiq** [comment]
maybe we should also offer tar downloads. On OSX I can create a tar on the command line and then simply double click to extract it from Finder.

**Comment by monkeyiq** [comment]
I am testing again on master, which uses the PHPZipStreamer library This extracts fine for me using unzip on Linux and the recommended "The Unarchiver" on OSX.

This is the manifest I have.
2017-11-13 07:25:56 .....     30965760     30965760  gb9b
2017-11-13 07:25:56 .....     10485760     10485760  gb9
2017-11-13 07:25:56 .....       104448       104448  gb8
2017-11-13 07:26:08 .....   1073741824   1073741824  gb5
2017-11-13 07:26:20 .....   1073741824   1073741824  gb4
2017-11-13 07:26:32 .....   1073741824   1073741824  gb3
2017-11-13 07:26:44 .....   1073741824   1073741824  gb2
2017-11-13 07:26:56 .....   1073741824   1073741824  gb1

7z can extract (though lists a header error) on Linux.

On windows 10 when I right click and extract the zip64 it all just works.

**Comment by monkeyiq** [comment]
So maybe we need some recommendation for windows 7, perhaps using 7-zip to extract?

**Comment by madsi1m** [comment]
Fixed in a1fff58c89c3a27288567e16b981e67a7ab08113
Cheers.
